ci/codeowners-validator/default.nix

```diff
@@ -0,0 +1,31 @@
+{
+  buildGoModule,
+  fetchFromGitHub,
+  fetchpatch,
+}:
+buildGoModule {
+  name = "codeowners-validator";
+  src = fetchFromGitHub {
+    owner = "mszostok";
+    repo = "codeowners-validator";
+    rev = "f3651e3810802a37bd965e6a9a7210728179d076";
+    hash = "sha256-5aSmmRTsOuPcVLWfDF6EBz+6+/Qpbj66udAmi1CLmWQ=";
+  };
+  patches = [
+    # https://github.com/mszostok/codeowners-validator/pull/222
+    (fetchpatch {
+      name = "user-write-access-check";
+      url = "https://github.com/mszostok/codeowners-validator/compare/f3651e3810802a37bd965e6a9a7210728179d076...840eeb88b4da92bda3e13c838f67f6540b9e8529.patch";
+      hash = "sha256-t3Dtt8SP9nbO3gBrM0nRE7+G6N/ZIaczDyVHYAG/6mU=";
+    })
+    # Undoes part of the above PR: We don't want to require write access
+    # to the repository, that's only needed for GitHub's native CODEOWNERS.
+    # Furthermore, it removes an unneccessary check from the code
+    # that breaks tokens generated for GitHub Apps.
+    ./permissions.patch
+    # Allows setting a custom CODEOWNERS path using the OWNERS_FILE env var
+    ./owners-file-name.patch
```

Comment:
Is there a reason you didn't merge at least the permissions patch (e.g. behind a flag) in your original PR?

Likewise for a PR upstream to the file path thing, though it is less of a concern given there's no overlap between your open PR and that specific functionality.

Reply:
Because the native CODEOWNERS requires people to have write access, the permission patch removing that check only makes sense when you don't use the native CODEOWNERS. The original purpose of the PR (https://github.com/mszostok/codeowners-validator/pull/222) was really just do do that write check, but here I'm mainly using it for just the side benefits of improved performance:

> Speeds up the users check by not querying for all organisation members (gets just a bit slow [if you have 3k members](https://github.com/orgs/NixOS/people)!), instead just query the users with access to the repository

Reply:
> Likewise for a PR upstream to the file path thing, though it is less of a concern given there's no overlap between your open PR and that specific functionality.

That would be good, I'm just a bit lazy :sweat_smile:

Reply:
Makes sense, thanks!